### ISSUE TITLE
Profanity shows old messages as new when connecting to some MUC

### ISSUE THREAD

**Issue by Misaflo** [profanity-im/profanity#1254]
## Expected Behavior

Old messages are not shown as new but are in the right place of the timeline.

## Current Behavior

In some MUC (on server running Openfire 4.5.0), after connect old messages are shown as just received with the current date and time (not in Profanity MUC).

## Context

Here some log (`profanity --log=DEBUG`):

    15/01/2020 21:29:11: xmpp: DBG: RECV: <message id="410757451959" to="me@domain.fr/profanity.8G1h" type="groupchat" from="muc@conference.domain.fr/other"><body>allez bonne soirée.</body><stanza-id id="71106678-f2e1-4054-89f7-17394a61b7af" by="muc@conference.domain.fr" xmlns="urn:xmpp:sid:0"/><delay xmlns="urn:xmpp:delay" stamp="2020-01-15T18:21:58.455Z" from="other@domain.fr/resource/></message>

There is: `stamp="2020-01-15T18:21:58.455Z"` (witch is the right hour, 18:21:58).
But in profanity I see the hour of the connection to the MUC (21:29:11).

This bug concern dev version (and does not occur in the stable release 0.7.1).

## Environment

    profanity --version    
    Profanity, version 0.7.1dev.makepkg.44204635
    Copyright (C) 2012 - 2019 James Booth <boothj5web@gmail.com>.
    Copyright (C) 2019 - 2020 Michael Vetter <jubalh@iodoru.org>.
    License GPLv3+: GNU GPL version 3 or later <https://www.gnu.org/licenses/gpl.html>
    
    This is free software; you are free to change and redistribute it.
    There is NO WARRANTY, to the extent permitted by law.
    
    Build information:
    XMPP library: libmesode
    Desktop notification support: Enabled
    OTR support: Enabled (libotr 4.1.1)
    PGP support: Enabled (libgpgme 1.13.1)
    OMEMO support: Enabled
    C plugins: Enabled
    Python plugins: Enabled (3.8.1)
    GTK icons: Enabled

Tested on Arch Linux and Debian Buster.

**Comment by jubalh** [comment]
@Misaflo can  you please test with latest master?

**Comment by Misaflo** [comment]
Yes, no change: the datetime is still wrong (datetime of the connection).

**Comment by jubalh** [comment]
@Misaflo can you test with latest master again?

**Comment by Misaflo** [comment]
No change :-(

**Comment by jubalh** [comment]
Can you please describe again what is happening and how you test this?

**Comment by jubalh** [comment]
Do you mean that messages that were sent while you were offline (to a MUC) are later displayed as regular messages and not as MUC history?

Or are you saying the date/time of all MUC messages is wrong?
Or the date/time of MUC history messages is wrong?

Also does this only happen to you on that openfire server?

**Comment by Misaflo** [comment]
I joining some MUC when connecting to my account.
On profanity MUC no problem. But on others (running up-to-date openfire), when I'm connecting I see all old messages as just received (the hour is wrong), even them that I have alreay read the day before.

For test this, I juste `/quit` profanity, et lanch it again (all messages are see as juste received, with current hour).

**Comment by jubalh** [comment]
` from="muc@conference.domain.fr/other` looks like MUC PM.

**Comment by jubalh** [comment]
https://xmpp.org/extensions/xep-0045.html#enter-history sais "The 'from' attribute MUST be set to the JID of the room itself."

> <message id="410757451959" to="me@domain.fr/profanity.8G1h" type="groupchat" from="muc@conference.domain.fr/other"><body>allez bonne soirée.</body><stanza-id id="71106678-f2e1-4054-89f7-17394a61b7af" by="muc@conference.domain.fr" xmlns="urn:xmpp:sid:0"/><delay xmlns="urn:xmpp:delay" stamp="2020-01-15T18:21:58.455Z" from="other@domain.fr/resource/></message>

has `from="other@domain.fr/resource/`.

Maybe an openfire problem?

**Comment by jubalh** [comment]
I created a bug for openfire: https://discourse.igniterealtime.org/t/maybe-muc-history-bug/87165

**Comment by jubalh** [comment]
Through this issue we fixed some problems anyways :)

1. We only added delay timestamps to messages if it was MUC history. Not regular delays. This was introduced in https://github.com/profanity-im/profanity/commit/61f66966ddfe8ebd8bae26dd7ff92d777004edda#diff-4926fd4577a336bd3eb240f8104a5c5bL837 and fixed in https://github.com/profanity-im/profanity/commit/8a9488245ba17e8040e85bdfbfd54463abeb99b3.

2. We sort the delay tags now and use the oldest.
